### PR TITLE
fix #22

### LINKED ISSUE #22: 2.3.0 crashes on Minecraft 1.19.3
I assume this was due to the Registry changes. Appears to be a 2 line fix

Crash when using in dev: https://gist.github.com/MoriyaShiine/5bdac92bad9ddfc3e91562b505f425b3

### PR DESCRIPTION
**Pull request by MoriyaShiine**
Updates some gradle (and etc) things, but more importantly fixes the crash introduced by 1.19.3. This version will only work on 1.19.3, and no longer works on 1.19.0-1.19.2.

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionSha256Sum=e6d864e3b5bc05cc62041842b306383fc1fefcec359e70cebb1d470a6094ca82
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.2-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.6-bin.zip
```
Comment (JamiesWhiteShirt) [inline gradle/wrapper/gradle-wrapper.properties]: I don't think this should be removed, otherwise IIRC the Gradle wrapper will complain about it (and for good reason, really). Gradle distributes checksums on their [web page](https://gradle.org/release-checksums/#v7.6). You may also have inadvertently changed from the complete distribution to the binary-only distribution. I find the complete distribution helps with debugging, though I don't really have a stake in this any more.

File: build.gradle [hunk 2]
```diff
@@ -2,7 +2,7 @@ import org.gradle.util.GradleVersion
 import java.time.Instant
 
 plugins {
-    id "fabric-loom" version "0.12.48"
+    id "fabric-loom" version "1.0-SNAPSHOT"
```
Comment (JamiesWhiteShirt) [inline build.gradle]: I don't really have a stake in this, but will comment regardless. I don't like depending on snapshots, especially for Loom. There is no need to be on the bleeding edge, and it is counter to the concept of reproducible builds. If you'd like to depend on more a specific version of Loom, the latest non-snapshot version at the time of writing is 1.0.15.

**Comment by MoriyaShiine** [comment]
Does this PR need anything else?

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,7 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionSha256Sum=e6d864e3b5bc05cc62041842b306383fc1fefcec359e70cebb1d470a6094ca82
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.2-all.zip
+distributionSha256Sum=312eb12875e1747e05c2f81a4789902d7e4ec5defbd1eefeaccc08acf096505d
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.6-all.zip
+networkTimeout=10000
```
Comment (ChloeDawn) [inline gradle/wrapper/gradle-wrapper.properties:5]: Why was this added to the wrapper properties?

File: src/main/resources/fabric.mod.json [hunk 4]
```diff
@@ -23,7 +23,7 @@
     "mixins.reach-entity-attributes.json"
   ],
   "depends": {
-    "minecraft": ">=1.19",
+    "minecraft": "~1.19.3",
```
Comment (ChloeDawn) [inline src/main/resources/fabric.mod.json:26]: How does `~` function in Fabric's versioning schema?

**Comment by ChloeDawn** [comment]
Apologies for the delay getting to this, it's been a busy few weeks.

File: gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (MoriyaShiine) [reply gradle/wrapper/gradle-wrapper.properties:5]: It's copied over from the example mod, but includes the checksum and uses complete gradle instead of binary-only

File: src/main/resources/fabric.mod.json (hunk 4 above)
Comment (MoriyaShiine) [reply src/main/resources/fabric.mod.json:26]: Anything above and including 1.19.3, and below and excluding 1.20

Edit, source: https://fabricmc.net/wiki/documentation:fabric_mod_json_spec?s[]=version

**Comment by MoriyaShiine** [comment]
No worries :+1:

**Comment by MoriyaShiine** [comment]
Are any changes needed?
